stumpy/gpu_aamp.py

```diff
@@ -10,6 +10,7 @@
 from numba import cuda
 
 from . import core, config
+from .core import _gpu_searchsorted_right
```

Comment:
I think this will be a problem. If you directly import from `.core` then your python environment will have no concept/idea of `_gpu_searchsorted_right_driver_not_found`! Instead, If there are no GPUs, it will still be trying to execute the real `core._gpu_searchsorted_right` 

The only way it will detect the "correct" `_gpu_searchsorted_right` is if you import it directly from `stumpy._gpu_searchsorted_right` (which inherits from `__init__.py`)

I will say that I don't love having a private function in the top-level `stumpy` namespace. I would actually prefer to continue being able to `from .core import _gpu_searchsorted_right` as you've shown and somehow trick the python environment into figuring out which function to use (either the GPU available one OR the `driver_not_found` version).

Reply:
Ohh! I think found a way!

In `__init__.py`, we do:

```
else:
    from . import core
    core._gpu_searchsorted_left = core._gpu_searchsorted_left_driver_not_found
```

Then, we might be able to keep our import as `from .core import _gpu_searchsorted_right`

Reply:
So, we'd remove `from .core import _gpu_searchsorted_right` in `__init__.py` and then simply replace the function when there is no GPU present

Reply:
This way, `_gpu_searchsorted_right` is NOT at the top-level import and can only be accessed via importing from `core`

Reply:
> If you directly import from `.core` then your python environment will have no concept/idea of `_gpu_searchsorted_right_driver_not_found`! Instead, If there are no GPUs, it will still be trying to execute the real `core._gpu_searchsorted_right`

Right! 

> I will say that I don't love having a private function in the top-level stumpy namespace. I would actually prefer to continue being able to from .core import _gpu_searchsorted_right as you've shown and somehow trick the python environment into figuring out which function to use (either the GPU available one OR the driver_not_found version).

Yeah... better to keep private functions in a private mode :)



> Ohh! I think found a way!
> 
> In `__init__.py`, we do:
> 
> ```
> else:
>     from . import core
>     core._gpu_searchsorted_left = core._gpu_searchsorted_left_driver_not_found
> ```
> 
> Then, we might be able to keep our import as `from .core import _gpu_searchsorted_right`

> This way, _gpu_searchsorted_right is NOT at the top-level import and can only be accessed via importing from core

That is a very very smart move! Cool!